usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/notice/warning/NonAsciiOrNonPrintableCharNotice.java

```diff
@@ -32,7 +32,7 @@ public NonAsciiOrNonPrintableCharNotice(String filename, String fieldName, Strin
                 "Non ascii or non printable character(s) in:" + idValue + " in field:"
                         + fieldName + " for entity with id:" + entityId);
 
-        this.entityId = entityId;
+        this.entityId = entityId != null ? entityId : "no id";
```

Comment:
To cover all cases
- make entityId private in base class
- add protected setEntityId that implement null check
- call this setter from derived classes constructor